compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/SkiaLayer.skiko.kt

```diff
@@ -186,27 +192,26 @@ internal class SkiaLayer(
         invalidate()
     }
 
-    // TODO(demin): support perspective projection for rotationX/rotationY (as in Android)
-    // TODO(njawad) Add camera distance leveraging Sk3DView along with rotationX/rotationY
-    // see https://cs.android.com/search?q=RenderProperties.cpp&sq= updateMatrix method
-    // for how 3d transformations along with camera distance are applied. b/173402019
     private fun updateMatrix() {
         val pivotX = transformOrigin.pivotFractionX * size.width
         val pivotY = transformOrigin.pivotFractionY * size.height
 
         matrix.reset()
+        matrix.translate(x = -pivotX, y = -pivotY)
         matrix *= Matrix().apply {
-            translate(x = -pivotX, y = -pivotY)
-        }
-        matrix *= Matrix().apply {
-            translate(translationX, translationY)
-            rotateX(rotationX)
-            rotateY(rotationY)
             rotateZ(rotationZ)
+            rotateY(rotationY)
+            rotateX(rotationX)
             scale(scaleX, scaleY)
         }
         matrix *= Matrix().apply {
-            translate(x = pivotX, y = pivotY)
+            // the camera location is passed in inches, set in pt
+            val depth = cameraDistance * 72f
```

Comment:
As I understand:
depth - in pixels.
cameraDistance - in inches.
72 - in pixels per inch (PPI). But this multiplier should depend on screen resolution and size.

Or 72 is a universal value for such cases?

Reply:
Or may depth - in Density pixels (dp)?

Reply:
The value and actually the comment picked up from [android implementation](https://cs.android.com/android/platform/superproject/+/master:external/skia/src/utils/SkCamera.cpp;l=181;drc=5ca657189aac546af0aafaba11bbc9c5d889eab3).

Initially, I thought that it should depend on density too. I event experimented with it in this branch (see commit history), but at the end - hardcoded `72` is the universal value to match android behavior.